core/shared/platform/include/platform_api_extension.h

```diff
@@ -977,6 +977,19 @@ os_socket_set_broadcast(bh_socket_t socket, bool is_enabled);
 int
 os_socket_get_broadcast(bh_socket_t socket, bool *is_enabled);
 
+/**
+ * Dump memory information of the current process
+ * It may have variant implementations in different platforms
+ *
+ * @param out the output buffer. It is for sure the return content
+ *            is a c-string which ends up with '\0'
+ * @param size the size of the output buffer
+ *
+ * @return 0 if success, -1 otherwise
+ */
+int
+os_dumps_proc_mem_info(char *out, unsigned int size);
```

Comment:
[minor] could we use `size_t` -like type for the `size` variable?

Reply:
Also, I think we could probably make the API more verbose - having `char *` type as a return value doesn't tell implementation and the user how the data should be structured (so on windows we might have different format, on linux there will be a different format).
I'd suggest using a well-defined structure that's being returned here so:
1. implementers know exactly what sort of data needs to be returned
2. users of that method know what to expect
3. we can have a consistent formatting defined in a common layer, rather than have different formatting for each system
4. If we ever want to emit metrics from WAMR or generate statistics, we can't rely on unstructured text.

This is just a suggestion - if returning an unstructured blob from this function is really what we want, that's fine, just wanted to raise the concern and make sure we're making a conscious decision.

Reply:
I am thinking that `os_dumps_proc_mem_info` should return some kind of *memory usage reports* of the current process. The developer should not be worried about how the memory data is structured or how different every platform is. It doesn't limit the output format and the output items.